tango/test_context.py

```diff
@@ -197,29 +229,63 @@ def generate_db_file(self, server, instance, device,
         """Generate a database file corresponding to the given arguments."""
         if not tangoclass:
             tangoclass = server
+        self.generate_db_file_tangoclass(server, instance, tangoclass)
+        device_prop_info = (
+            {
+                "name": device,
+                "properties": properties
+            }
+        )
+        return self.generate_db_file_device(device_prop_info)
+
+    def generate_db_file_tangoclass(self, server, instance, tangoclass):
+        """Generate a database file corresponding to the given arguments.
+
+        Only device server and device class information (no devices information)
+        """
         # Open the file
-        with open(self.db, 'w') as f:
+        with open(self.db, "a") as f:
             f.write("/".join((server, instance, "DEVICE", tangoclass)))
-            f.write(': "' + device + '"\n')
+            f.flush()
+
+    def generate_db_file_device(self, device_prop_info):
+        """Generate a database file corresponding to the given arguments.
+
+        Only devices information (neither device server nor device class
+        information)
+        """
+        # Open the file
+        device_names = [info["name"] for info in device_prop_info]
+        with open(self.db, "a") as f:
+            for device_name in device_names:
+                f.write(': "' + device_name + '"\n')
```

Comment:
I'm still wondering about this?

Reply:
> I don't know how to reply to this: #314 (comment) so I generate a new comment here...
>
> What do you exactly mean? Whether the MultiDeviceTestContext.append_db_file method name is fine? or maybe its API? Or you reconsider having two methods as it was before #314 (comment).

I'm thinking about the code that will parse this file.  My question is about the format of the line with `DEVICE` in it.  The current version of your `append_db_file` method will have the first device name on the same line as the `DEVICE` part.  Then the next device is on a new line:
```
Clock/clock/DEVICE/Clock: "test/nodb/clock1"
: "test/nodb/clock2"
```

It is just a bit inconsistent, so I was wondering if it works.  Something like this would be more "consistent" for me:
```
Clock/clock/DEVICE/Clock
: "test/nodb/clock1"
: "test/nodb/clock2"
```

I tried exporting some properties from Jive to see the file format, but the files didn't have a `DEVICE` line.

Reply:
Sorry I didn't see your initial comment. Now I understand your concern.
Actually now testing it with two devices of the same class I have realized that it does now work correctly. I will investigate it tomorrow.

Reply:
The update in 7c7a649 for multiple devices look good.  Thanks.